### PR TITLE
Add support for multiple resource classes

### PR DESCRIPTION
**Pull request by KhoiCaoSaritasa**
- add support for multiple resource classes
- refactor `CeleryExportAdminMixin`, `CeleryImportAdminMixin`: 
    - move common logic to a base class
    - drop `get_export_formats`, `get_import_formats`: we now support multiple resource classes, and each resource supports different set of formats, so those formats should be retrieved by calling the corresponding methods on resource class

### CODE REVIEW

File: import_export_extensions/admin/mixins/base.py [hunk 1]
```diff
@@ -35,3 +63,131 @@ def app_label(self):
...
+                    self.model,
+                    resource_class=resources.CeleryModelResource,
+                ),
+            ]
+        if self.resource_classes:
+            return self.resource_classes
+        return [self.resource_class]
+
+    def get_resource_kwargs(
+        self, request, *args, **kwargs,
```
Comment (NikAzanov) [inline import_export_extensions/admin/mixins/base.py:118]: Put each argument on the new line
```suggestion
        self,
        request,
        *args,
        **kwargs,
```

File: import_export_extensions/admin/mixins/base.py [hunk 2]
```diff
@@ -35,3 +63,131 @@ def app_label(self):
...
+    ) -> dict[str, typing.Any]:
+        """Return common resource kwargs."""
+        return {}
+
+    def get_resource_index(self, form) -> int:
```
Comment (NikAzanov) [inline import_export_extensions/admin/mixins/base.py:123]: Would be nice to add type hint to `form` argument

File: import_export_extensions/admin/mixins/base.py [hunk 3]
```diff
@@ -35,3 +63,131 @@ def app_label(self):
...
+                resource_index = int(form.cleaned_data["resource"])
+            except ValueError:
+                pass
+        return resource_index
+
+
+class BaseImportMixin(BaseImportExportMixin):
+    """Base class for import mixin."""
+
+    def get_import_resource_classes(self) -> typing.Sequence[ResourceType]:
```
Comment (NikAzanov) [inline import_export_extensions/admin/mixins/base.py:137]: > "be conservative in what you send, be liberal in what you accept" - Postel's law (robustness principle)

```suggestion
    def get_import_resource_classes(self) -> list[ResourceType]:
```

File: import_export_extensions/static/import_export_extensions/js/admin/export_form.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+(function($) {
+  $(document).ready(function() {
+    const exportForm = $('#export-form');
```
Comment (NikAzanov) [inline import_export_extensions/static/import_export_extensions/js/admin/export_form.js:3]: Please, unify quotes, use `"`

**Review by NikAzanov — CHANGES_REQUESTED** [review]
So let's get rid of deprecated `resource_class` logic. It already deprecated in original package, but we have a chance just remove it because our package wasn't released yet.

File: tests/settings.py [hunk 5]
```diff
@@ -97,6 +100,12 @@
     },
 ]
 
+
+# Configure `drf-spectacular` to check it works for import-export API
+REST_FRAMEWORK = {
+    'DEFAULT_SCHEMA_CLASS': 'drf_spectacular.openapi.AutoSchema',
```
Comment (OttoAndrey) [inline tests/settings.py:106]:
```suggestion
    "DEFAULT_SCHEMA_CLASS": "drf_spectacular.openapi.AutoSchema",
```

**Pull request closed without merging**